dsm/resource_sobject.go

```diff
@@ -304,6 +308,16 @@ func createSO(ctx context.Context, d *schema.ResourceData, m interface{}) diag.D
 		}
 	}
 
+	if d.Get("fpe").(string) != "" && d.Get("fpe_radix").(int) != 0 {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "only one of these two can be given in the terraform configuration: fpe, fpe_radix",
```

Comment:
Shouldn't the backend throw this error? Why duplicate the validation logic?

Reply:
The above validation is not to validate the post body of DSM key creation. Rather, it is to ensure that the new release does not create issues for users who are currently using support for "fpe_radix"

1. In previous version, there was support only for radix in fpeOptions, and the end user could specify radix in "fpe_radix" object.
2. In this release, we expect the end user to define all fpeOptions in the "fpe" object.
3. Because we added support for radix earlier than we did for all fpeOptions, we cannot expect the end user to reconfigure their terraform processes. For that reason, we decided that we will let fpe_radix support remain like that. The new users of fpe support will only use "fpe" object. And the customers who asked for "fpe_radix" will continue to use it without changing anything in their configuration. 
4. Ideally, we do not want the customer to specify radix separately, everything should be specified in the "fpe" object itself.

Reply:
So, you're implementing backwards compatibility here on the client side, is that right?
If so, you need to update these messages to reflect that, and you need to document that fact in the code and the module docs.